src/caret_analyze/architecture/architecture.py

```diff
@@ -466,6 +466,41 @@ def rename_topic(self, src: str, dst: str) -> None:
         for c in self._communications:
             c.rename_topic(src, dst)
 
+    @staticmethod
+    def diff_node_names(
```

Comment:
doc string must be described (for each function).

Reference:
https://github.com/tier4/CARET_analyze/blob/f17cd5b74af6d12398e356809368dade5e856675/src/caret_analyze/architecture/architecture.py#L397-L408